# HG changeset patch
# User Michael Goffioul <michael.goffioul@gmail.com>
# Date 1367884914 14400
#      Mon May 06 20:01:54 2013 -0400
# Node ID 96d02808a8819439feb98dee0c3a06de26bd3fd1
# Parent  9c0487bb205b6616a9064d9835ab939312aaab5c
Support "true" or "yes" for %NO_OCT_FILE_STRIP% in mkoctfile.in.cc.

* src/mkoctfile.in.cc (is_true): New function.
(main): Use it on %NO_OCT_FILE_STRIP% replacement.

diff --git a/src/mkoctfile.in.cc b/src/mkoctfile.in.cc
--- a/src/mkoctfile.in.cc
+++ b/src/mkoctfile.in.cc
@@ -391,16 +391,23 @@ ends_with (const string& s, const string
 static int
 run_command (const string& cmd)
 {
   if (debug)
     cout << cmd << endl;
   return system (cmd.c_str ());
 }
 
+bool
+is_true (const std::string& s)
+{
+  return (s == "yes"
+          || s == "true");
+}
+
 int
 main (int argc, char **argv)
 {
   initialize ();
 
   string file, output_option;
   list<string> cfiles, ccfiles, f77files;
   int result = 0;
@@ -409,17 +416,17 @@ main (int argc, char **argv)
   string libfiles = "";
   string octfile = "";
   string outputfile = "";
   string incflags = "";
   string defs = "";
   string ldflags = "";
   string pass_on_options = "";
   bool strip = false;
-  bool no_oct_file_strip_on_this_platform = %NO_OCT_FILE_STRIP%;
+  bool no_oct_file_strip_on_this_platform = is_true ("%NO_OCT_FILE_STRIP%");
   bool link = true;
   bool link_stand_alone = false;
   string output_ext = ".oct";
   bool depend = false;
   bool compile = true;
 
   if (argc == 1)
     {
